### DISCUSSION TITLE
Preserve song in playlist on song upgrade

### DISCUSSION THREAD

**Discussion by alex-phillips** [navidrome/navidrome#2323] [Ideas]
I'm not sure how involved this would be, but it would be great if songs that exist in playlists are preserved when the song is upgraded. I've been going through and re-ripping existing CDs I have to FLAC when previously they were maybe 256. The issue I'm running in to is I have playlists that have been built up over time, but as as I an upgrading these songs / albums to better quality, they are removed from playlists because the song file no longer exists. This gets becomes very problematic as while I'm upgrading, I don't know if my playlists are slowly deteriorating over time.

I don't know how involved this is. I have noticed that Plex has this functionality. Perhaps it's storing a reference to a musicbrainz song ID in the playlist as well and when a song gets upgraded, it still has that same musicbrainz ID associated with it?

**Comment by deluan** [comment]
Yeah, this is caused by the way the scanner works today. There are plans to change the way we identify songs/albums/artists, that will solve this in a future version. The issue to follow is this one: #1976 (comment)
